File: libraries/managed-exceptions-lib/src/python/managed_exceptions/authorization/unauthenticated_exception.py
from http import HTTPStatus
from managed_exceptions.error_details import ErrorDetails
from managed_exceptions.managed_exception import ManagedException

class UnauthenticatedException(ManagedException):
    def __init__(self, message: str, diagnostic_details: dict[str, str] = {}):
        super().__init__(ErrorDetails(
            status_code=HTTPStatus.UNAUTHORIZED,
            diagnostic_code="00401",
            diagnostic_details=diagnostic_details,
            message=message
        ))

File: services/Sphana/Libraries/Sphana.ManagedException/Authorization/UnauthenticatedException.cs
using System.Net;

namespace Sphana.ManagedException.Authorization;

/// <summary>
/// Thrown when a request lacks valid authentication credentials.
/// </summary>
public class UnauthenticatedException : ManagedException
{
    public UnauthenticatedException(string message, Dictionary<string, string>? diagnosticDetails = null)
        : base(HttpStatusCode.Unauthorized, "UNAUTHENTICATED", message, diagnosticDetails)
    {
    }
}